tests/components/ambient_network/test_sensor.py

```diff
@@ -17,65 +18,47 @@
 from tests.common import async_fire_time_changed, snapshot_platform
 
 
-@freeze_time("2023-11-08")
-@pytest.mark.parametrize("config_entry", ["AA:AA:AA:AA:AA:AA"], indirect=True)
+@freeze_time("2023-11-9")
+@pytest.mark.parametrize(
+    "config_entry",
+    ["AA:AA:AA:AA:AA:AA", "CC:CC:CC:CC:CC:CC", "DD:DD:DD:DD:DD:DD"],
+    indirect=True,
+)
 @pytest.mark.usefixtures("entity_registry_enabled_by_default")
 async def test_sensors(
     hass: HomeAssistant,
     open_api: OpenAPI,
-    aioambient,
-    config_entry,
+    aioambient: AsyncMock,
+    config_entry: ConfigEntry,
     entity_registry: er.EntityRegistry,
     snapshot: SnapshotAssertion,
 ) -> None:
     """Test all sensors under normal operation."""
     await setup_platform(True, hass, config_entry)
-
     await snapshot_platform(hass, entity_registry, snapshot, config_entry.entry_id)
 
 
-@freeze_time("2023-11-09")
-@pytest.mark.parametrize("config_entry", ["AA:AA:AA:AA:AA:AA"], indirect=True)
-async def test_sensors_with_stale_data(
-    hass: HomeAssistant, open_api: OpenAPI, aioambient, config_entry
-) -> None:
-    """Test that the sensors are not populated if the data is stale."""
-    await setup_platform(False, hass, config_entry)
-
-    sensor = hass.states.get("sensor.station_a_absolute_pressure")
-    assert sensor is None
-
-
-@freeze_time("2023-11-08")
 @pytest.mark.parametrize("config_entry", ["BB:BB:BB:BB:BB:BB"], indirect=True)
 async def test_sensors_with_no_data(
-    hass: HomeAssistant, open_api: OpenAPI, aioambient, config_entry
+    hass: HomeAssistant,
+    open_api: OpenAPI,
+    aioambient: AsyncMock,
+    config_entry: ConfigEntry,
 ) -> None:
     """Test that the sensors are not populated if the last data is absent."""
-    await setup_platform(False, hass, config_entry)
-
-    sensor = hass.states.get("sensor.station_b_absolute_pressure")
-    assert sensor is None
-
-
-@freeze_time("2023-11-08")
-@pytest.mark.parametrize("config_entry", ["CC:CC:CC:CC:CC:CC"], indirect=True)
-async def test_sensors_with_no_update_time(
-    hass: HomeAssistant, open_api: OpenAPI, aioambient, config_entry
-) -> None:
-    """Test that the sensors are not populated if the update time is missing."""
-    await setup_platform(False, hass, config_entry)
+    await setup_platform(True, hass, config_entry)
 
-    sensor = hass.states.get("sensor.station_c_absolute_pressure")
-    assert sensor is None
+    sensor = hass.states.get("sensor.station_b_temperature")
+    assert sensor is not None
+    assert "last_measured" not in sensor.attributes
```

Comment:
Can you please add also a test, where you assert that `last_measured` has the correct value

Reply:
`last_measured` is captured in the snapshot.